src/algebraic_geometry/AffineScheme.lean

```diff
@@ -307,4 +307,94 @@ begin
     exact congr_arg subtype.val (X.map_prime_spectrum_basic_open_of_affine x).symm }
 end
 
+/-- The prime ideal of `𝒪ₓ(U)` corresponding to a point `x : U`. -/
+noncomputable
+def is_affine_open.prime_ideal_of {X : Scheme} {U : opens X.carrier}
+  (hU : is_affine_open U) (x : U) :
+  prime_spectrum (X.presheaf.obj $ op U) :=
+((Scheme.Spec.map (X.presheaf.map (eq_to_hom $
+  show U.open_embedding.is_open_map.functor.obj ⊤ = U, from
+    opens.ext (set.image_univ.trans subtype.range_coe)).op).op).1.base
+  ((@@Scheme.iso_Spec (X.restrict U.open_embedding) hU).hom.1.base x))
+
+lemma is_affine_open.from_Spec_prime_ideal_of {X : Scheme} {U : opens X.carrier}
+  (hU : is_affine_open U) (x : U) :
+  hU.from_Spec.val.base (hU.prime_ideal_of x) = x.1 :=
+begin
+  dsimp only [is_affine_open.from_Spec, subtype.coe_mk],
+  erw [← Scheme.comp_val_base_apply, ← Scheme.comp_val_base_apply],
+  simpa only [← functor.map_comp_assoc, ← functor.map_comp, ← op_comp, eq_to_hom_trans, op_id,
+    eq_to_hom_refl, category_theory.functor.map_id, category.id_comp, iso.hom_inv_id_assoc]
+end
+
+lemma is_affine_open.is_localization_stalk_aux {X : Scheme} (U : opens X.carrier)
+  [is_affine (X.restrict U.open_embedding)] :
+  (inv (Γ_Spec.adjunction.unit.app (X.restrict U.open_embedding))).1.c.app
+    (op ((opens.map U.inclusion).obj U)) =
+      X.presheaf.map (eq_to_hom $ by rw opens.inclusion_map_eq_top :
+        U.open_embedding.is_open_map.functor.obj ⊤ ⟶
+          (U.open_embedding.is_open_map.functor.obj ((opens.map U.inclusion).obj U))).op ≫
+      to_Spec_Γ (X.presheaf.obj $ op (U.open_embedding.is_open_map.functor.obj ⊤)) ≫
+      (Scheme.Spec.obj $ op $ X.presheaf.obj $ _).presheaf.map
+        (eq_to_hom (by { rw [opens.inclusion_map_eq_top], refl }) : unop _ ⟶ ⊤).op :=
+begin
+  have e : (opens.map (inv (Γ_Spec.adjunction.unit.app (X.restrict U.open_embedding))).1.base).obj
+    ((opens.map U.inclusion).obj U) = ⊤,
+  by { rw [opens.inclusion_map_eq_top], refl },
+  rw [Scheme.inv_val_c_app, is_iso.comp_inv_eq, Scheme.app_eq _ e,
+    Γ_Spec.adjunction_unit_app_app_top],
+  simp only [category.assoc, eq_to_hom_op],
+  erw ← functor.map_comp_assoc,
+  rw [eq_to_hom_trans, eq_to_hom_refl, category_theory.functor.map_id,
+    category.id_comp],
+  erw Spec_Γ_identity.inv_hom_id_app_assoc,
+  simp only [eq_to_hom_map, eq_to_hom_trans],
+end
+
+lemma is_affine_open.is_localization_stalk {X : Scheme} {U : opens X.carrier}
+  (hU : is_affine_open U) (x : U) :
+  @is_localization (X.presheaf.obj $ op U) _ (hU.prime_ideal_of x).as_ideal.prime_compl
+    (X.presheaf.stalk x) _ _ :=
```

Comment:
There is `localization.away`, which is specifically for localization at prime ideals. Can you use that here?